config/reference_config.go

```diff
@@ -148,10 +149,23 @@ func (c *ReferenceConfig) Refer(_ interface{}) {
 
 		// TODO(decouple from directory, config should not depend on directory module)
 		if regUrl != nil {
-			cluster := extension.GetCluster("registryAware")
+			// for multi-subscription scenario, use 'zone-aware' policy by default
+			cluster := extension.GetCluster(cluster_impl.GetZoneAwareName())
+			// The invoker wrap sequence would be:
+			// ZoneAwareClusterInvoker(StaticDirectory) ->
+			// FailoverClusterInvoker(RegistryDirectory, routing happens here) -> Invoker
 			c.invoker = cluster.Join(directory.NewStaticDirectory(invokers))
 		} else {
-			cluster := extension.GetCluster(c.Cluster)
+			// not a registry url, must be direct invoke.
+			clu := cluster_impl.GetFailoverName()
+			if len(invokers) > 0 {
+				u := invokers[0].GetUrl()
+				if nil != &u {
+					clu = u.GetParam(constant.CLUSTER_KEY, cluster_impl.GetZoneAwareName())
+				}
+			}
+
+			cluster := extension.GetCluster(clu)
 			c.invoker = cluster.Join(directory.NewStaticDirectory(invokers))
```

Comment:
i think this two logic can move out of `if-else`

Reply:
I think do following changes: 
```go
		var hitClu string
		if regUrl != nil {
			// for multi-subscription scenario, use 'zone-aware' policy by default
			hitClu = constant.ZONEAWARE_CLUSTER_NAME
		} else {
			// not a registry url, must be direct invoke.
			hitClu = constant.FAILOVER_CLUSTER_NAME
			if len(invokers) > 0 {
				u := invokers[0].GetUrl()
				if nil != &u {
					hitClu = u.GetParam(constant.CLUSTER_KEY, constant.ZONEAWARE_CLUSTER_NAME)
				}
			}
		}

		cluster := extension.GetCluster(hitClu)
		// If 'zone-aware' policy select, the invoker wrap sequence would be:
		// ZoneAwareClusterInvoker(StaticDirectory) ->
		// FailoverClusterInvoker(RegistryDirectory, routing happens here) -> Invoker
		c.invoker = cluster.Join(directory.NewStaticDirectory(invokers))
```